### PR TITLE
Adding prompt for delete connection 

### LINKED ISSUE #19764: Confirmation prompt on "Delete Connection"
When right-clicking on a server from the server list, and you want to refresh the connection, it's easy to accidentally click "Delete Connection", and if you do, theres no confirmation prompt.

image

I suggest following options to avoid these kind of accidents in the future.
* When you click on "Delete Connection", a confirmation prompt will appear
* Refresh and delete options are separated away from each other
* Move "Delete Connection" to the "Edit Connection" drawer.

### PR DESCRIPTION
**Pull request by aasimkhan30**
This PR fixes #19764
image

### CODE REVIEW

File: src/sql/platform/connection/test/common/testNotificationService.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { INotification, INotificationHandle, INotificationService, IPromptChoice, IPromptChoiceWithMenu, IPromptOptions, IStatusMessageOptions, NotificationMessage, NotificationsFilter, Severity } from 'vs/platform/notification/common/notification';
+import { Event, IDisposable } from 'vs/workbench/workbench.web.main';
+
+export class TestNotificationService implements INotificationService {
```
Comment (Charles-Gagnon) [inline src/sql/platform/connection/test/common/testNotificationService.ts:9]: This already exists - src\vs\platform\notification\test\common\testNotificationService.ts

File: src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts [hunk 2]
```diff
@@ -79,6 +81,34 @@ suite('SQL Connection Tree Action tests', () => {
...
+		let notificationService = TypeMoq.Mock.ofType<INotificationService>(TestNotificationService, TypeMoq.MockBehavior.Loose);
+		notificationService.callBase = true;
+		notificationService.setup(x => x.prompt(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns((
+			severity: Severity,
+			message: string,
+			choices: IPromptChoice[],
+			options?: IPromptOptions
+		) => {
+			choices[promptChoice].run();
+			return {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts:94]: Would be simpler to just use NoOpNotification from src\vs\platform\notification\common\notification.ts

**Review by kburtram — APPROVED** [review]
Hmm, not sure I love the UX as most confirmation prompts tend to be modal in most products.  But I guess this is what we have in the product to use so is fine.

File: src/sql/workbench/services/objectExplorer/browser/connectionTreeAction.ts [hunk 3]
```diff
@@ -283,10 +285,37 @@ export class DeleteConnectionAction extends Action {
...
+				}, {
+					label: deleteConnectionConfirmationNo,
+					run: async () => {
+					}
+				}]);
 		} else if (this.element instanceof ConnectionProfileGroup) {
-			await this._connectionManagementService.deleteConnectionGroup(this.element);
+			this._notificationService.prompt(Severity.Warning,
+				localize('deleteConnectionGroupConfirmation', "Are you sure you want to delete connection group {0}?", this.element.name),
+				[{
```
Comment (kburtram) [inline src/sql/workbench/services/objectExplorer/browser/connectionTreeAction.ts:308]: maybe add quote characters around the connection name?

**Comment by Charles-Gagnon** [comment]
There is a modal dialog - you use the `IDialogService.show` for that (we do it for prompting about running tasks when shutting down for example)

File: src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts [hunk 4]
```diff
@@ -79,6 +81,17 @@ suite('SQL Connection Tree Action tests', () => {
 		return connectionManagementService;
 	}
 
+	function createDialogService(choiceIndex: number): TypeMoq.Mock<IDialogService> {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts:84]: Add doc comments - it's not clear what "choiceIndex" will do without reading through the code.

File: src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts [hunk 5]
```diff
@@ -329,22 +342,54 @@ suite('SQL Connection Tree Action tests', () => {
...
+			saveProfile: true,
+			id: 'testId'
+		});
+		let connectionAction: DeleteConnectionAction = new DeleteConnectionAction(DeleteConnectionAction.ID,
+			DeleteConnectionAction.DELETE_CONNECTION_LABEL,
+			connection,
+			connectionManagementService.object,
+			createDialogService(1).object); // Selecting 'No' on the modal dialog
+
+		return connectionAction.run().then((value) => {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts:378]: Would suggest just making this async

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**